harmoni-web-api/src/main/java/org/harmoniapp/harmoniwebapi/services/UserService.java

```diff
@@ -122,53 +126,51 @@ public UserDto add(UserDto userDto) {
 
         ContractType contractType = repositoryCollector.getContractTypes()
                 .findById(userDto.contractType().getId())
-                .orElseThrow(IllegalArgumentException::new);
-
-        user.setContractType(contractType);
-
-        User supervisor = repositoryCollector.getUsers().findById(userDto.supervisorId()).orElse(null);
-        user.setSupervisor(supervisor);
-
-        Address residence;
-        if (existingUser.isPresent()) {
-            residence = existingUser.get().getResidence();
-            residence = addressService.updateAddress(residence, userDto.residence());
-        } else {
-            residence = addressService.saveAddressEntity(userDto.residence());
-        }
-        user.setResidence(residence);
-
-        Address workAddress;
-        if (existingUser.isPresent()) {
-            workAddress = existingUser.get().getWorkAddress();
-            workAddress = addressService.updateAddress(workAddress, userDto.workAddress());
-        } else {
-            workAddress = addressService.saveAddressEntity(userDto.workAddress());
-        }
-        user.setWorkAddress(workAddress);
-
-        user.setLanguages(
+                .orElseThrow(() -> new IllegalArgumentException("Contract Type with ID " + id + " not found"));
+
+        User supervisor = repositoryCollector.getUsers().findById(userDto.supervisorId())
+                .orElse(null);
+
+        existingUser.setFirstname(userDto.firstname());
+        existingUser.setSurname(userDto.surname());
+        existingUser.setEmail(userDto.email());
+        existingUser.setContractType(contractType);
+        existingUser.setSupervisor(supervisor);
+        existingUser.setContractSignature(userDto.contractSignature());
+        existingUser.setContractExpiration(userDto.contractExpiration());
+        existingUser.setPhoneNumber(userDto.phoneNumber());
+        existingUser.setEmployeeId(userDto.employeeId());
+
+        Address residence = existingUser.getResidence();
+        residence = addressService.updateAddress(residence, userDto.residence());
+        existingUser.setResidence(residence);
+
+        Long newDepartmentId = userDto.workAddress().id();
+        Address newDepartment = repositoryCollector.getAddresses().findById(newDepartmentId)
+                .orElseThrow(() -> new IllegalArgumentException("Department with ID " + id + " not found"));
+        existingUser.setWorkAddress(newDepartment);
+
+        existingUser.setLanguages(
                 userDto.languages().stream()
                         .map(p -> repositoryCollector.getLanguages().findById(p.id()).get())
                         .collect(Collectors.toSet()));
 
-        user.setRoles(
+        existingUser.setRoles(
                 userDto.roles().stream()
                         .map(p -> repositoryCollector.getRoles().findById(p.getId()).get())
                         .collect(Collectors.toSet())
         );
```

Comment:
Assuming that you always get a full userDto object, when converted to an entity class, you can assign the object ID from the DB instead of assigning the field values one by one, in this case you will only be left with the setting of fields that require additional DB queries and password. This is possible by first overriding the equals(Object o) method in the entity class